`=` in f-string interpolation causes false positives and negatives for `invalid-assignment`

### Summary

The `=` part of f-string interpolations is ignored, causing false positives and false negatives for `invalid-assignment`. [Example](https://play.ty.dev/8aab7fa7-d811-481d-8084-16b956a9cbc0):
```console
$ cat >invalid-assignment.py <<'# EOF'
from typing import Literal
fp: Literal["1=1"] = f"{1=}"  # false positive
fn: Literal["2"] = f"{2=}"  # false negative
# EOF

$ ty check --output-format concise invalid-assignment.py
invalid-assignment.py:2:22: error[invalid-assignment] Object of type `Literal["1"]` is not assignable to `Literal["1=1"]`
Found 1 diagnostic
```

### Version

ty 0.0.5 (d37b7dbd9 2025-12-20)

--- a/crates/ty_python_semantic/resources/mdtest/literal/f_string.md
+++ b/crates/ty_python_semantic/resources/mdtest/literal/f_string.md
@@ -24,6 +24,16 @@ string = "hello"
 reveal_type(f"{string!r}")  # revealed: str
 ```
 
+## Debug Specifier
+
+The `=` specifier causes the expression text and value to be included in the output:
+
+```py
+# f"{1=}" evaluates to "1=1", but we fall back to `str` for now
+reveal_type(f"{1=}")  # revealed: str
+reveal_type(f"value: {42=}")  # revealed: str
+```
+
 ## Format Specifiers
 
 ```py

--- a/crates/ty_python_semantic/src/types/infer/builder.rs
+++ b/crates/ty_python_semantic/src/types/infer/builder.rs
@@ -7594,7 +7594,7 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
                                     range: _,
                                     node_index: _,
                                     expression,
-                                    debug_text: _,
+                                    debug_text,
                                     conversion,
                                     format_spec,
                                 } = expression;
@@ -7613,7 +7613,10 @@ impl<'db, 'ast> TypeInferenceBuilder<'db, 'ast> {
                                 // (`Type::format`?) that handles the `__format__` method.
                                 // Conversion flags should be handled before calling `__format__`.
                                 // https://docs.python.org/3/library/string.html#format-string-syntax
-                                if !conversion.is_none() || format_spec.is_some() {
+                                if debug_text.is_some()
+                                    || !conversion.is_none()
+                                    || format_spec.is_some()
+                                {
                                     collector.add_expression();
                                 } else {
                                     if let Type::StringLiteral(literal) = ty.str(self.db()) {

